pkg/pillar/hypervisor/pci.go

```diff
@@ -93,6 +93,15 @@ func (d pciDevice) vid() (string, error) {
 	return strings.TrimSpace(strings.TrimSuffix(string(vendorID), "\n")), nil
 }
 
+// read device ID
+func (d pciDevice) devid() (string, error) {
```

Comment:
I understand how this works, and see that we use it in `kubevirt.go`. Why is it needed now where it wasn't before? What changes with kubevirt about how we manage PCI devices vs others?

Reply:
Look at kubevirt.go -> registerWithKV() I added comments with links to kubevirt documentation.

// Register the host device with Kubevirt
// Refer https://kubevirt.io/user-guide/virtual_machines/host-devices/#host-preparation-for-pci-passthrough
// Refer https://kubevirt.io/user-guide/virtual_machines/host-devices/#usb-host-passthrough

We need to register the PCIe devices with kubevirt before using those for PCIe passthrough. And kubevirt expects the PCIHostDevices config to be passed in PCIVendorSelector form which is nothing but vendorid:devid

Reply:
> We need to register the PCIe devices with kubevirt before using those for PCIe passthrough. And kubevirt expects the PCIHostDevices config to be passed in PCIVendorSelector form which is nothing but vendorid:devid

How does that work when a server has two or more identical PCI cards i.e., identical vendorid:devid?
We see this with Ethernet cards on some servers.
So are you sure we can't use the bus:port.function way to identify a PCIe device to kubevirt?

Reply:
Not as far as I know. If you look at this document
https://kubevirt.io/user-guide/virtual_machines/host-devices/#host-preparation-for-pci-passthrough

"Currently, the KubeVirt device plugin doesn't allow the user to select a specific device by specifying the address. Therefore, if multiple NVMe devices with the same vendor and product id exist in the cluster, they could be randomly assigned to a VM. If the devices are not on the same node, then the nodeSelector mitigates the issue."


I do not see any other way than PCIVendorSelector.